lib/Sema/TypeCheckAttr.cpp

```diff
@@ -115,20 +115,18 @@ class AttributeChecker : public AttributeVisitor<AttributeChecker> {
     auto &C = D->getASTContext();
 
     if (isa<DestructorDecl>(D)) {
+      if (!C.LangOpts.hasFeature(Feature::IsolatedDeinit)) {
+        diagnoseAndRemoveAttr(attr, diag::isolated_deinit_experimental);
+        return;
+      }
+
       if (auto nominal = dyn_cast<NominalTypeDecl>(D->getDeclContext())) {
         if (!isa<ClassDecl>(nominal)) {
           // only classes and actors can have isolated deinit.
           diagnoseAndRemoveAttr(attr, diag::isolated_deinit_on_value_type);
           return;
         }
       }
-
-      TypeChecker::checkAvailability(
```

Comment:
Similar comment to above: should this be kept around for if the experimental feature is enabled?